test/Microsoft.Azure.EventHubs.Tests/Processor/ProcessorTestBase.cs

```diff
@@ -948,6 +948,30 @@ async Task SingleProcessorHostWithAadTokenProvider()
             await RunGenericScenario(eventProcessorHost, epo);
         }
 
+        [Fact]
+        [DisplayTestMethodName]
+        async Task ReRegisterEventProcessor()
+        {
+            var eventProcessorHost = new EventProcessorHost(
+                null, // Entity path will be picked from connection string.
+                PartitionReceiver.DefaultConsumerGroupName,
+                TestUtility.EventHubsConnectionString,
+                TestUtility.StorageConnectionString,
+                Guid.NewGuid().ToString());
+
+            // Calling register for the first time should succeed.
+            TestUtility.Log("Registering EventProcessorHost for the first time.");
+            await eventProcessorHost.RegisterEventProcessorAsync<TestEventProcessor>();
+
+            // Unregister event processor should succed
+            TestUtility.Log("Registering EventProcessorHost for the first time.");
+            await eventProcessorHost.UnregisterEventProcessorAsync();
+
+            // Register other event processor
+            TestUtility.Log("Registering other Event Processor in the same host after unregister should succed.");
+            await eventProcessorHost.RegisterEventProcessorAsync<SecondTestEventProcessor>();
```

Comment:
>await eventProcessorHost.RegisterEventProcessorAsync<SecondTestEventProcessor>(); [](start = 12, length = 81)

Run generic scenario instead of this. Generic scenario will ensure receives are also working OK.

await RunGenericScenario(eventProcessorHostSecond);

Reply:
Sure will do :)